packages/slice-machine/server/src/api/slices/delete.ts

```diff
@@ -0,0 +1,158 @@
+  const desiredLibrary = libraries.find((library) => library.name === libName);
+  if (!desiredLibrary) {
+    const message = `When deleting slice: ${sliceId}, the library: ${libName} was not found.`;
+    console.error(`[slice/delete] ${message}`);
+    return {
+      err: new Error(message),
+      status: 500,
+      reason: message,
+      type: "error",
+    };
+  }
+  const desiredSlice = desiredLibrary.components.find(
+    (component) => component.model.id === sliceId
+  );
+
+  if (!desiredSlice) {
+    const message = `When deleting slice: ${sliceId}, the slice: ${sliceId} was not found.`;
+    console.error(`[slice/delete] ${message}`);
+    return {
+      err: new Error(message),
+      status: 500,
+      reason: message,
+      type: "error",
+    };
+  }
+
+  const sliceDirectory = CustomPaths(env.cwd)
+    .library(libName)
+    .slice(desiredSlice.model.name);
+
+  const generatedSliceDirectory = GeneratedPaths(env.cwd)
+    .library(libName)
+    .slice(desiredSlice.model.name);
+
+  try {
+    IO.Slice.deleteSlice(sliceDirectory.value());
+  } catch (err) {
+    console.error(`[slice/delete] ${err as string}`);
+    return {
+      err: err,
+      reason: "We couldn't delete your slice. Check your terminal.",
+      status: 500,
+      type: "error",
+    };
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  const deleteAssets = async () => {
+    try {
+      IO.Slice.deleteSlice(generatedSliceDirectory.value());
+    } catch (err) {
+      console.error(
+        `[slice/delete] Could not delete your slice assets files.\n`,
+        `To resolve this, manually delete the directory ${generatedSliceDirectory.value()}`
+      );
+      throw err;
+    }
+  };
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  const updateMockConfig = async () => {
+    try {
+      removeSliceFromMockConfig(req.env.cwd, {
+        key: desiredSlice.model.name,
+        prefix: libName,
+      });
+    } catch (err) {
+      console.error(
+        `[slice/delete] Could not delete your slice from the mock-config.json.\n`,
+        `To resolve this, manually remove the ${
+          desiredSlice.model.name
+        } field in ${MocksConfig(req.env.cwd)}`
+      );
+      throw err;
+    }
+  };
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  const updateTypes = async () => {
+    try {
+      IO.Types.upsert(env);
+    } catch (err) {
+      console.error(
+        `[slice/delete] Could not update the project types.\n`,
+        `You can manually delete these in ${resolve(
+          path.join(env.cwd, ".slicemachine", "prismicio.d.ts")
+        )}`
+      );
+      throw err;
+    }
+  };
+
+  const updateLibraries = async () => {
+    try {
+      await onSaveSlice(env);
+    } catch (err) {
+      console.error(
+        `[slice/delete] Could not update the slice libraries.\n`,
+        `You can manually resolve this in ....`
```

Comment:
Need to figure out how this log will work. not sure we actually want people editing these files?